vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -199,45 +200,49 @@ public Retrofit getRetrofit() {
         return mRetrofit;
     }
 
+    /**
+     * @return the current VimeoAccount, or null if it hasn't been set yet
+     */
+    @Nullable
     public VimeoAccount getVimeoAccount() {
-        if (mVimeoAccount == null) {
-            throw new AssertionError("Account should never be null");
-        }
-
         return mVimeoAccount;
     }
 
+    public boolean isAuthenticated() {
+        return mVimeoAccount != null && mVimeoAccount.isAuthenticated();
+    }
+
     public void setVimeoAccount(@Nullable VimeoAccount vimeoAccount) {
         if (vimeoAccount == null) {
-            // If the provided account was null but we have an access token, persist the vimeo account with
-            // just a token in it. Otherwise we'll want to leave the persisted account as null.
-            vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
             if (mConfiguration.accessToken != null) {
-                mConfiguration.saveAccount(vimeoAccount, null);
+                // If the provided account was null but we have an access token, persist the vimeo account with
+                // just a token in it. Otherwise we'll want to leave the persisted account as null.
+                vimeoAccount = new VimeoAccount(mConfiguration.accessToken);
+                mConfiguration.saveAccount(vimeoAccount);
             }
         }
 
         mVimeoAccount = vimeoAccount;
     }
 
     /**
-     * @deprecated use {@link #saveAccount(VimeoAccount, String)} instead
-     * <p/>
-     * We find no use in storing the password when you can persist the {@link VimeoAccount} across
-     * application sessions.
-     */
-    @Deprecated
-    public void saveAccount(@Nullable VimeoAccount vimeoAccount, String email, String password) {
-        saveAccount(vimeoAccount, email);
-    }
-
-    /**
      * Sets the {@link #mVimeoAccount} field as well as triggering the saveAccount event for the
-     * account store
+     * account store.
+     *
+     * @param vimeoAccount The account to save - this should be provided; if it is not, then one will
+     *                     be constructed using the access token set on the {@link Configuration}.
+     * @param accountName  accountName should be provided if the account is for an authenticated user. It may
+     *                     be null if the account represents a client credentials grant.
      */
-    public void saveAccount(@Nullable VimeoAccount vimeoAccount, String email) {
+    public void saveAccount(@Nullable VimeoAccount vimeoAccount, @Nullable String accountName) {
```

Comment:
Maybe it makes sense to split this into two versions like it is in Configuration?